crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -288,290 +353,280 @@ impl AppHandlerExecute for Ics20Transfer {
         mut state: S,
         msg: &MsgTimeout,
     ) -> anyhow::Result<()> {
-        // we put source and dest as chain_a (the source) as we're refunding tokens,
-        // and the destination chain of the refund is the source.
-        execute_ics20_transfer(
-            &mut state,
-            &msg.packet.data,
-            &msg.packet.port_on_a,
-            &msg.packet.chan_on_a,
-            &msg.packet.port_on_a,
-            &msg.packet.chan_on_a,
-            true,
-        )
-        .await
-        .context("failed to refund tokens during timeout_packet_execute")
+        refund_tokens(&mut state, &msg.packet)
+            .await
+            .context("failed to refund tokens during timeout_packet_execute")
     }
 
+    #[instrument(skip_all)]
     async fn acknowledge_packet_execute<S: StateWrite>(
         mut state: S,
         msg: &MsgAcknowledgement,
     ) -> anyhow::Result<()> {
-        let ack: TokenTransferAcknowledgement = serde_json::from_slice(
-            msg.acknowledgement.as_slice(),
-        )
-        .expect("valid acknowledgement, should have been checked in acknowledge_packet_check");
-        if ack.is_successful() {
-            return Ok(());
+        let ack: TokenTransferAcknowledgement =
+            serde_json::from_slice(msg.acknowledgement.as_slice())
+                .context("failed to deserialize token transfer acknowledgement")?;
+        if !ack.is_successful() {
```

Comment:
This just flips the logic: instead of returning early on success, success is now the fall-through case while failure triggers a refund.